app/src/main/java/com/example/habitsmasher/HabitEvent.java

```diff
@@ -0,0 +1,96 @@
+package com.example.habitsmasher;
+
+import android.net.Uri;
+
+import com.google.firebase.firestore.PropertyName;
+
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * This is the Habit Event class
+ * Its purpose is to store and retrieve the start date, comments, picture, and location of a habit event
+ */
+public class HabitEvent {
+    private Date _startDate;
+    private String _comment;
+    private Uri _pictureUri;
+    private UUID _id = UUID.randomUUID();
+
+    // TODO: Eventually add location
+
+    /**
+     * Empty constructor
+     */
+    public HabitEvent() {
+        this._startDate = new Date();
+        this._comment = "";
+        // TODO: Set default image
+    }
+
+    /**
+     * Default constructor, creates new HabitEvent
+     * @param startDate (Date): The start date of the habit event
+     * @param comment (String): The comment of the habit event
+     * @param pictureUri (String): The URL of the picture of the habit event
+     */
+    public HabitEvent(Date startDate, String comment, Uri pictureUri) {
+        this._startDate = startDate;
```

Comment:
The `this` keyword isn't needed here (or in **any** of the getters/setters in this class). Since our fields start with an `_`, we don't need to differentiate the variable context.

If we named our fields the same as our local variables, then we would do:
`this._startDate = startDate`
But, since they are named **differently**, we can do:
`_startDate = startDate`

The `this` keyword is already **implied**.